### PR TITLE
Issue #1253 helpers to resolve packages and bindings

### PR DESCRIPTION
**Pull request by ioana-blue**
Add methods for resolving packages and retrieving all intermediary bindings. Refactor code to use these methods. No semantic change intended

### CODE REVIEW

**Comment by ioana-blue** [comment]
@rabbah after some iterations, only one helper function left as discussed.

File: common/scala/src/main/scala/whisk/core/entity/WhiskPackage.scala [hunk 1]
```diff
@@ -140,6 +148,20 @@ object WhiskPackage
     val bindingFieldName = "binding"
     override val collectionName = "packages"
 
+    /**
+     * an utility function that traverses all potential binding references and returns the resolved package
+     * @param pkg the package for which the resolution to be applied
+     * @return the same package if there is no binding, or the fully resolved package
+     */
+    def resolveBinding(entityStore: EntityStore, pkg: DocId)(
```
Comment (rabbah) [inline common/scala/src/main/scala/whisk/core/entity/WhiskPackage.scala]: Isn't there a layer on top of this which given a sequence of strings resolves those into fully qualified names?

Comment (ioana-blue) [inline common/scala/src/main/scala/whisk/core/entity/WhiskPackage.scala]: It depends on how it is used. For one action, there will not be such a sequence. For a sequence of actions, you'll have such a layer.

**Comment by ioana-blue** [comment]
I think I will close this and incorporate it in the work for #1274 that actually uses it. It's a small code change and better have it together with its usage.

**Pull request closed without merging**